src/rpc/server.cpp

```diff
@@ -550,6 +550,51 @@ std::string experimentalDisabledHelpMsg(const std::string& rpc, const std::vecto
         + config;
 }
 
+std::string asOfHeightMessage(bool hasMinconf) {
+    std::string minconfInteraction = hasMinconf
+        ? "                    `minconf` must be at least 1 when `asOfHeight` is provided.\n"
+        : "";
+    return
+        ". asOfHeight       (numeric, optional, default “*”) Execute the query as if it\n"
+        "                    were run when the blockchain was at the height specified by\n"
+        "                    this argument. The default is to use the entire blockchain\n"
+        "                    that the node is aware of. A “future” height will fall back\n"
+        "                    to the current height. Any explicit value will cause the\n"
+        "                    mempool to be ignored, meaning no unconfirmed tx will be\n"
+        "                    considered.\n"
+        + minconfInteraction;
+}
+
+std::optional<int> parseAsOfHeight(const UniValue& params, int index) {
+    std::optional<int> asOfHeight;
+    if (params.size() > index && !(params[index].isStr() && params[index].get_str() == "*")) {
+        auto requestedHeight = params[index].get_int();
+        if (requestedHeight < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Can not perform the query as of a negative block height");
```

Comment:
Other height parameters accept a negative argument, for example [getblock](https://zcash.github.io/rpc/getblock.html):
```
1. "hash|height"          (string, required) The block hash or height. Height can be negative where -1 is the last known valid block
```
This would allow the default to be "-1" rather than defining a special "*" value.